File: haskell/2022/Day1.hs
module Day1 where

import Data.List
import Data.List.Split
import Data.Ord

main' :: IO ()
main' = do
  r <- take 3 . sortBy (comparing Data.Ord.Down) . map (sum . map read) . splitWhen null . lines <$> readFile "../inputs/2022/Day1/input.txt"
  -- part 1
  print $ head r
  -- part 2
  print $ sum $ take 3 r

File: elixir/2022/lib/day1.ex
defmodule Day1 do
  def main() do
    content =
      File.read!("../../inputs/2022/Day1/input.txt")
      |> String.trim_trailing("\n")
      |> String.split("\n\n")
      |> Enum.map(&String.split(&1, "\n"))

    sorted =
      content
      |> Stream.map(fn m ->
        m
        |> Stream.map(&Integer.parse(&1))
        |> Stream.map(&elem(&1, 0))
        |> Enum.sum()
      end)
      |> Enum.sort(:desc)

    part1 =
      sorted
      |> Enum.take(1)

    IO.inspect(part1)

    part2 =
      sorted
      |> Enum.take(3)
      |> Enum.sum()

    IO.inspect(part2)
  end
end
